lib/util.js

```diff
@@ -15,9 +16,42 @@ module.exports.opensslHashFunctionName = (algorithm) => {
   return algorithms.getAlgorithm(algorithm, op)[op]();
 };
 
-module.exports.limitUsages = (usages, allowed) => {
+module.exports.limitUsages = (usages, allowed, err = SyntaxError) => {
   for (const usage of usages) {
     if (!allowed.includes(usage))
-      throw new SyntaxError();
+      throw new err();
   }
 };
+
+module.exports.decodeBase64Url = (enc) => {
```

Comment:
this is an alternative b64 encoding? maybe you can link to the spec for it?

Reply:
Defined in https://tools.ietf.org/html/rfc4648#section-5

And throughout JOSE rfc family applied with no padding.

Reply:
Yeah, I never came across it before either. Super weird stuff.

Reply:
Ref: https://github.com/panva/jose/blob/master/lib/help/base64url.js

Reply:
I also considered using `replace`, but I assumed that using three regular expressions just to substitute single characters cannot possibly be efficient. On the other hand, right now I am splitting the string into a character array, mapping it onto a different array and creating a new string out of it, so my approach also seems terribly inefficient. (Great candidate for premature optimization!)